dora-parser/src/parser.rs

```diff
@@ -93,75 +93,85 @@ impl<'a> Parser<'a> {
         &mut self,
         elements: &mut Vec<Elem>,
     ) -> Result<(), ParseErrorAndPos> {
-        let modifiers = self.parse_annotations()?;
+        let annotation_usages = self.parse_annotation_usages()?;
 
         match self.token.kind {
             TokenKind::Fun => {
-                self.restrict_modifiers(
-                    &modifiers,
+                self.reject_annotation_usages(
+                    &annotation_usages,
                     &[
-                        Modifier::Internal,
-                        Modifier::Optimize,
-                        Modifier::OptimizeImmediately,
-                        Modifier::Test,
-                        Modifier::Cannon,
+                        self.interner.intern("abstract"),
+                        self.interner.intern("override"),
+                        self.interner.intern("open"),
+                        self.interner.intern("final"),
+                        self.interner.intern("pub"),
+                        self.interner.intern("static"),
```

Comment:
We should probably intern these things at the start once and keep the names somewhere to refer to them when needed.

Reply:
We should do this before merging.

Reply:
Agreed! What would be the best place for it?